proxy/ParentSelection.cc

```diff
@@ -742,7 +742,7 @@ ParentRecord::Init(matcher_line *line_info)
     unavailable_server_retry_responses = new UnavailableServerResponseCodes(nullptr);
   }
 
-  if (this->parents == nullptr && go_direct == false) {
+  if (this->parents == nullptr || go_direct == false) {
```

Comment:
I'm not familiar with this area at all, but this change doesn't look correct.

The doc says:
> Every line in the parent.config file must contain either a parent= or go_direct= directive.

So the condition is checking either of them is configured, I think.

Reply:
Agreed.

Reply:
@maskit and @zwoop, I think this code is correct as is.  It prevents you from making a mistake in the parent.config.  If the parent list is null and you set go_direct to false any transaction matching that config would 502 as there are no parents to forward to and go_direct false prevents you from going to the origin directly.